content/articles/graphs-in-data-structure -using _c++/index.md

```diff
@@ -0,0 +1,153 @@
+
+A tree cannot have any loops or cycles, whereas graphs may.
+
+![graph](./graph_5.jpg)
+
+### Graph Representation 
+A graph can be represented in several ways. The following are the two most frequent ways of expressing a graph:
+
+* **Adjacency Matrix** 
+A V*V binary matrix is an adjacency matrix.There is an edge that is connecting vertex i and vertex j, element A<sub>i,j</sub> is 1, otherwise A<sub>i,j</sub> is 0.
+
+  **Note**:A binary matrix has cells that can only have one of two possible values: 0 or 1.
+
+We can store the cost of the edge at A<sub>i,j</sub> in the adjacency matrix of a weighted graph rather than storing 0 or 1.
+
+If A<sub>i,j</sub> is 1 in undirected graph then A<sub>i,j</sub> will also be 1.
+
+If A<sub>i,j</sub> is 1 in  directed graph, then may or may not be 1.
+
+The adjacency matrix offers constant-time access (O(1)) to detect if two nodes have an edge. The adjacency matrix has a space complexity of O(V)<sup>2</sup>.
+
+* **Adjacency List**
+It is a most common way of representing a graph when we implement it in our code. An adjacency list is a collection of distinct lists in an array A. Every node of a graph A<sub>i</sub> is a list which hold all the values of the vertices next to vertex i.
+
+If we have a weighted graph we store the cost with the vertex using pairs you will get better understanding when we implement it in our code. If vertex j is in list A<sub>i</sub>, vertex i will be in list A<sub>j</sub> in an undirected graph.
+
+Due to the fact that information in the adjacency list is only stored for edges that exist in the graph, its space complexity is O(V + E).
+Representation of an undirected graph,
+
+* G is an undirected graph with 5 vertices and 7 edges.
+  
+  ![graph](./graph_6.jpg)
+
+* Adjacency-list representation of G
+  
+  ![graphList](./graph_7.jpg)
+
+* The adjacency-matrix representation of G
+  
+  ![graphMartrix](./graph_8.jpg)
+
+**Code for Adjacency list representation of graph** : 
+
+``` C++
+#include <iostream>
+#include<list>
+using namespace std;
+
+class graph{
+public:
+    list<int> *adjlist;
+    int n;
+    graph(int v){
+        adjlist=new list<int> [v];
+        n=v;
+    }
+
+    void addedge(int u,int v,bool bi){
+        adjlist[u].push_back(v);
+        if(bi){
+            adjlist[v].push_back(u);
+        }
+    }
+
+    void print(){
+        for(int i=0;i<n;i++){
+            cout<<i<<"-->";
+            for(auto it:adjlist[i]){
+                cout<<it<<" ";
+            }
+            cout<<endl;
+        }
+        cout<<endl;
+    }
+
+
+};
+
+int main() {
+    graph g(5);
+    g.addedge(1,2,true);
+    g.addedge(4,2,true);
+    g.addedge(1,3,true);
+    g.addedge(4,3,true);
+    g.addedge(1,4,true);
+
+    g.print();
+}
+
+```
+
+The complete execution of the above code can be found [here](https://replit.com/@AnubhavBansal1/Gaphs#main.cpp).
```

Comment:
Change to " You can find the complete execution of the above code [here](https://replit.com/@AnubhavBansal1/Gaphs#main.cpp)."